airbyte-db/src/main/java/io/airbyte/db/AirbyteVersion.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Airbyte
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.db;
+
+/**
+ * The AirbyteVersion identifies the version of the database used internally by Airbyte services.
+ */
+public class AirbyteVersion implements Comparable<AirbyteVersion> {
+
+  public static final String AIRBYTE_VERSION_KEY_NAME = "airbyte_version";
+
+  private final String version;
+  private final String major;
+  private final String minor;
+  private final String patch;
+
+  public AirbyteVersion(final String version) {
+    this.version = version;
+    final String[] parsedVersion = version.replace("\n", "").strip().split("-")[0].split("\\.");
+    String major = "";
+    String minor = "";
+    String patch = "";
+    if (parsedVersion.length >= 1)
+      major = parsedVersion[0];
+    if (parsedVersion.length >= 2)
+      minor = parsedVersion[1];
+    if (parsedVersion.length >= 3)
+      patch = parsedVersion[2];
+    this.major = major;
+    this.minor = minor;
+    this.patch = patch;
+  }
+
+  public String getVersion() {
+    return version;
+  }
+
+  public String getMajorVersion() {
+    return major;
+  }
+
+  public String getMinorVersion() {
+    return minor;
+  }
+
+  public String getPatchVersion() {
+    return patch;
+  }
+
+  /**
+   * Compares two Airbyte Version to check if they are equivalent.
+   *
+   * Only the major and minor part of the Version is taken into account.
+   */
+  @Override
+  public int compareTo(final AirbyteVersion another) {
+    if (version.equals("dev") || another.version.equals("dev"))
+      return 0;
```

Comment:
Not sure if we want to allow airbyte dev versions easy access to start or not...